app/services/poi_importer.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class PoiImporter
+  def import_from_csv(path_to_csv, kind)
+    CSV.foreach(path_to_csv) do |row|
+      if find_poi(row[0], row[1])
+        puts 'Already exist'
+      else
+        create(row[0], row[1], kind)
+      end
+    end
+  end
+
+  def import_all_files(path)
+    Dir.foreach(path) do |file|
+      next if ['.', '..', '.DS_Store'].include? file
+      if find_kind(file).is_a?(Hash)
+        puts "# Bad filename: #{file} or kind not found. Use: rake motohelper:import_poi['#{path}#{file}, ..kind..'] #"
+        next
+      end
+      puts "Importing #{file} :"
+      import_from_csv(path + file, find_kind(file))
+    end
+  end
+
+  private
+
+  def find_kind(file_name)
+    Workshop.kinds.each_key do |kind|
+      return kind if file_name.downcase.include? kind
+    end
+  end
+
+  def find_poi(longitude, latitude)
+    Workshop.find_by(longitude: longitude, latitude: latitude)
+  end
+
+  def create(longitude, latitude, kind)
+    if longitude.nil? || latitude.nil?
+      puts 'Not imported: missing attributes'
+    else
+      Workshop.create(kind: kind, longitude: longitude, latitude: latitude)
```

Comment:
The best way to import multi records is creating a big INSERT sql statement and execute it once. 
```sql
INSERT INTO films (code, title, did, date_prod, kind) VALUES
    ('B6717', 'Tampopo', 110, '1985-02-10', 'Comedy'),
    ('HG120', 'The Dinner Game', 140, DEFAULT, 'Comedy');
```
Important: Beware SQL Injection
```ruby
# Unsafe
st = ActiveRecord::Base.connection.raw_connection.prepare(
  "select * from users where email = '#{email}'")
results = st.execute
st.close

# Safe
st = ActiveRecord::Base.connection.raw_connection.prepare(
  "select * from users where email = ?")
results = st.execute(email)
st.close
```
In our case field 'kind' should be parameterization.
http://gavinmiller.io/2015/fixing-sql-injection-vulnerabilities/

Reply:
EDIT: if you dont' want use SQL, you can try adding multiple records using ActiveRecords 'create'.
``` person.pets.create([{name: 'Spook'}, {name: 'Choo-Choo'}]) ```
Instead a SQL you may build an array with hashes. I don't know if this will work. The key is that the all record should be add in ONE transaction. Maybe it's wort to try :)